onyo/commands/edit.py

```diff
@@ -18,39 +18,56 @@
-    return ["git -C \"" + git_directory + "\" commit -m", "\'edit " + file + "\'"]
+def build_commit_cmd(files, git_directory):
+    return ["git -C \"" + git_directory + "\" commit -m", "edit files\n" + "\n".join(files)]
 
 
-def edit(args):
-    # set paths
-    git_directory = get_git_root(args.file)
-    full_filepath = get_full_filepath(git_directory, args.file)
-    # path relative from git root dir
-    git_filepath = os.path.relpath(full_filepath, git_directory)
-
-    # check if file is in git
-    run_output = run_cmd("git -C \"" + git_directory + "\" ls-tree -r HEAD ")
-    if git_filepath not in run_output:
-        logger.error(git_filepath + " is not in onyo.")
+def check_sources(sources):
+    problem_str = ""
+    list_of_sources = []
+    if isinstance(sources, str):
+        sources = ["".join(sources)]
+    git_directory = get_git_root(sources[0])
+    for source in sources:
+        test_git = get_git_root(source)
+        current_source = get_full_filepath(test_git, source)
+        if git_directory != test_git:
+            problem_str = problem_str + "\n" + current_source + " not in same git as " + sources[0]
+        if not os.path.exists(current_source):
+            problem_str = problem_str + "\n" + current_source + " does not exist."
+        else:
+            list_of_sources.append(current_source)
+    if problem_str != "":
+        logger.error(problem_str + "\nNo folders or assets moved.")
         sys.exit(1)
+    return list(dict.fromkeys(list_of_sources))
 
-    # change file
-    if not args.non_interactive:
-        edit_file(full_filepath)
-
-    # check if changes happened and add+commit them
-    repo = Repo(git_directory)
-    changedFiles = [item.a_path for item in repo.index.diff(None)]
-    if len(changedFiles) != 0:
-        # build commands
-        git_add_cmd = build_git_add_cmd(git_directory, git_filepath)
-        [commit_cmd, commit_msg] = build_commit_cmd(git_filepath, git_directory)
-        # run commands
-        run_cmd(git_add_cmd)
-        run_cmd(commit_cmd, commit_msg)
-    elif args.non_interactive:
-        logger.info("no changes made.")
-    else:
-        logger.error("no changes made.")
-        sys.exit(1)
+
+def edit(args):
+    # check and set paths
+    list_of_sources = check_sources(args.file)
+
+    for source in list_of_sources:
+        git_directory = get_git_root(source)
+        git_filepath = os.path.relpath(source, git_directory)
+
+        # check if file is in git
+        run_output = run_cmd("git -C \"" + git_directory + "\" ls-tree -r HEAD ")
+        if git_filepath not in run_output:
+            logger.error(git_filepath + " is not in onyo.")
+            sys.exit(1)
+
+        # change file
+        if not args.non_interactive:
+            edit_file(source)
+
+        # check if changes happened and add them
+        repo = Repo(git_directory)
+        changed_files = [item.a_path for item in repo.index.diff(None)]
+        if len(changed_files) != 0:
```

Comment:
Is it possible to have a negative length (error state or something?) If so, the check should be `> 0`.

Reply:
changed_files should be a normal list, so I don't think that could happen.
But if it would happen, I am equally not sure, if we wanted to "go on and add the changes".
It might be worth trying to break this part, and if we find something adjust the behaviour, but I think I will leave it like that until we do harder tests to break all layers of onyo.

Reply:
Seems like `len()` is ok, and will only return > 0.

https://stackoverflow.com/questions/42521449/how-does-python-ensure-the-return-value-of-len-is-an-integer-when-len-is-cal